### DISCUSSION TITLE
Cannot mount external HDD anymore

### DISCUSSION THREAD

**Discussion by albotroz** [blakeblackshear/frigate#14929] [Ask A Question] [answered]
`Cannot mount bind_frigate at /data/media/frigate because it is not empty`

After resetting my router with attached HDD (but forgot to unplug or shutoff Rpi4 with HAOS-Frigate) I cannot add my external HDD anymore into Frigate.
There was /media/frigate folder and I've deleted it but it is still not possible.

**Comment by hawkeye217** [comment]
This Docker error typically means you're trying to mount a volume or bind mount to a directory inside the container that already contains files or directories. This isn't specific to Frigate itself.

**Comment by albotroz** [comment]
And how do I fix this without completely reinstalling and setting up HAOS?

**Reply by mimainka** [reply, marked as answer]
Id do it like this:

1. STOP frigate
2. ssh into haos, go to /data/media/frigate --> cd /data/media/frigate
3. check whats inside with ls command
4. move all the stuff inside this folder somewhere else
5. mkdir /data/media/frigate2 for example
6. mv /data/media/frigate/* /data/media/frigate2/
7. check if folder frigate is empty and everything is moved to frigate2 with ls for both directories
8. mount your network storage again under frigate
9. mv everything from frigate2 to frigate --> mv /data/media/frigate2/* /data/media/frigate/
10. Delete the frigate2 folder --> rm -rf /data/media/frigate2
11. Start frigate

Little Edit:
Of course it depends on the files that are in frigate and how much time went by since your hdd is not mounted anymore.
I guess after you lost connection to the network share, frigate created the directories directly inside of HAOS in that folder.
So if you want to transfer (and thus override) your physical /data/media/frigate-folder from HAOS or not depends on what data you want to have.

You will have an old db on your external hdd for example in comparison with whats in your HAOS-frigate folder.
If you changed the config after you lost connection, the new config will also only be in your HAOS-folder.

So personally I would use the config, models and db from the HAOS-folder (and not the ones from your hdd) if the time between "no hdd and now" is a few days.
If you are just having a few hours: simply use rm -rf inside /data/media/frigate and continue with the mount procedure. But definitely stop frigate before you try to delete files and start it again after you mounted your hdd.

**Reply by albotroz** [reply]
I've tried it now like you described it with creating another folder, renaming it.
This worked.
Thank you. This procedure should be in the docs for this problem. Because I have noticed that users of HAOS keep having this problem.

**Reply by mimainka** [reply]
It’s basically a problem that only occurs in a scenario where you loose connection with a network share. 
the only real HA contact point here is backups. 
the rest is addons that write into those places and HA is not responsible for them. 
so I guess that’s why nothing is written down about it. 
but ive been there too when my HA started a backup while my OpenMediaVault wasn’t running and it created stuff in the very same folder. 

Great to hear that it finally works for you now.

**Comment by albotroz** [comment]
So I have now taken out the SD card several times / deleted the media folder manually (SSH not set up - except for SSH addon, which is useless for anything anyway because it has no access to host). Also deleted frigate from the mount folder.
Frigate deactivated before I shut down Rpi so that media/frigate folder is not created again - HAOS created media folder after reboot (what a suprise...).
Ok, tried to connect the hard disk again with disabled Frigate.

Mounting HDD did not succeed. `Check host logs for errors from mount or systemd unit mnt-data-supervisor-mounts-frigate.mount for details`

Wasted so much time since yesterday just because of this problem. Am thinking of restoring backup, but probably it is as reliable as HomeAssistant updates... so not at all.

**Reply by mimainka** [reply]
So let me get a few answers first:
You are running HAOS directly on a RPI and you are using the frigate addon within HA-OS correct?
Whats the Supervisor log showing regarding your mounting attempt?

I've seen your other "Issue" reported inside the HA Github regarding "nothing works". Is your HA working again?

Your claim about the SSH Addon not allowing you to delete the folders under the /media/ section is wrong by the way. 
While you cannot go down to host-level, your shares are mounted inside the containers (so exactly where you have access with it).
Both work: The "Avanced SSH & Web Terminal" (even w safe mode enabled) and the "Terminal & SSH".
Empyting the frigate folder from the either addon MUST be enough to not run into the error regarding a non empty directory.

And Im not wanting to be mean, but: your writing transports a little attitude ("useless for anything, reliable [...] not at all") towards actually pretty well working stuff in case users are not messing with it.

**Reply by albotroz** [reply]
I really appreciate your help/answers and from everyone else who helps.

HAOS (not Frigate itself) keeps driving me crazy though.

I'll try again as you suggested.

Edit: About the SSH add-on. I can indeed see and delete /media folders. I did that yesterday and it looked like it was deleted.
Then today I take out the SD card and see the folder exists and also still with the old recordings with over 6GB. Which shouldn't actually be there.
Hence my statement about the SSH addon.

Yes HAOS ist working again. But only after I deleted the sqlite database from homeassistant. Restarted a thousand times, tried to make updates etc. An update (core) is still not possible. But the HomeAssistant team has already announced this.

Now the only thing left is the HDD.
